src/vulkan/buffer_descriptor.cc

```diff
@@ -74,7 +74,11 @@ Result BufferDescriptor::CreateResourceIfNeeded() {
 }
 
 Result BufferDescriptor::MoveResourceToBufferOutput() {
-  Result r = BufferBackedDescriptor::MoveResourceToBufferOutput();
+  Result r;
+
+  if (!IsReadOnly())
+    r = BufferBackedDescriptor::MoveResourceToBufferOutput();
+
```

Comment:
I think it would make more sense to do this in `BufferBackedDescriptor::MoveResourceToBufferOutput()`, as we check IsReadOnly() in the other methods of BufferBackedDescriptor.